### PR TITLE
DAOS-623 packaging: Fix go/golang per distro name

### PR DESCRIPTION
**Pull request by brianjmurrell**
In the Requires: on the client-tests

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
  * `Test-tag: test_dfuse_daos_build_wb` was used to validate that the daos build test passes
 * [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — COMMENTED** [review]
Thanks for this. I just hit this exact problem on Aurora! Does this not need a release bump and changelog?

**Comment by brianjmurrell** [comment]
> Thanks for this. I just hit this exact problem on Aurora! Does this not need a release bump and changelog?

It does, once I have seen that it's successful and I am ready for it to be reviewed and landed.  But I also need to add the test tag that does the daos build test.  Not sure what it is yet though.  Will have to go digging for it.

**Comment by ashleypittman** [comment]
Can you update the component in the PR title.

**Comment by brianjmurrell** [comment]
> Can you update the component in the PR title.

I would suggest that `packaging` should be a very valid component.  That's exactly what is being updated here.  Nothing else.

But I don't remember where these components come from.

**Comment by ashleypittman** [comment]
> > Can you update the component in the PR title.
> 
> I would suggest that `packaging` should be a very valid component.  That's exactly what is being updated here.  Nothing else.
> 
> Do I just add it to https://github.com/daos-stack/daos/blob/6e55d3085200ec38549c1fdcd975d647a58be014/ci/jira_query.py#L30-L31 or is there something else that uses these component names?

Yes, add it to there if you think it's valid. The check is run using the target, not the pr so this pr will still warn about it but future ones should be happy.

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -554,7 +554,10 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
-* Thu Jun 29 2023 Michael MacDonald <mjmac.macdonald@intel.com> 2.3.105-6
+* Fri Jul 07 2023 Brian J. Murrell <brian.murrell@intel.com> 2.5.100-7
+- Fix golang aos-client-tests dependency to be go instead
```
Comment (daltonbohning) [inline utils/rpms/daos.spec:558]:
```suggestion
- Fix golang daos-client-tests dependency to be go instead
```

**Review by daltonbohning — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**